lua/nvim_lsp/rust_analyzer.lua

```diff
@@ -1,11 +1,86 @@
 local configs = require 'nvim_lsp/configs'
 local util = require 'nvim_lsp/util'
+local server_name = 'rust_analyzer'
+local release_url = 'https://api.github.com/repos/rust-analyzer/rust-analyzer/releases/latest'
+local postfix = '-linux'
+local bin = 'rust-analyzer'
+
+if vim.fn.has('mac') then
+  postfix = '-mac'
+elseif vim.fn.has('win32') or vim.fn.has('win64') then
+  postfix = '-windows.exe'
+end
+
+local function make_installer()
+  local I = {}
```

Comment:
why not just use a `do` block with local variables, instead of this OOP thing?

Reply:
Do you have an example of what you mean? I'm pretty new to lua? I was looking at the [elixirLS](https://github.com/neovim/nvim-lsp/blob/master/lua/nvim_lsp/elixirls.lua#L17) file for inspiration.